tests/jballerina-unit-test/src/test/resources/test-src/types/regexp/regexp_value_test.bal

```diff
@@ -81,6 +81,12 @@ function testRegExpValueWithCharacterClass() {
 
     string:RegExp x7 = re `[\p{sc=Latin}\p{gc=Lu}\p{Lt}\tA\)]??`;
     assertEquality("[\\p{sc=Latin}\\p{gc=Lu}\\p{Lt}\\tA\\)]??", x7.toString());
+    
+    string:RegExp x8 = re `abc[a-z\]A-Z${"abc"}]`;
+    assertEquality("abc[a-z\\]A-Z${\"abc\"}]", x8.toString());
+    
+    string:RegExp x9 = re `abc\[a-z\]A-Z${"abc"}`;
+    assertEquality("abc\\[a-z\\]A-Zabc", x9.toString());
```

Comment:
Add a test for interpolation inside escaped char class
```
re `\[${a}\]
```

Reply:
Fixed in [#93b4](https://github.com/ballerina-platform/ballerina-lang/pull/40607/commits/93b48347d303014f83739b447f179a30a28a4792).